index.js

```diff
@@ -3,15 +3,30 @@
 module.exports = (a = [], b) => {
 	const data = typeof a === 'function' ? [] : Array.isArray(a) ? a : [a]
 	let queue = []
-	let out = (typeof a === 'function' ? a : b) || (d => queue.push(d))
+	let out = (typeof a === 'function' ? a : b) || toQueue
 
 	const log = create(data)
 	log.out = fn => {
-		out = fn
+		if (fn === undefined) throw new Error('`out` function is `undefined`')
+
+		if (fn === null) {
+			queue = []
```

Comment:
This would mean emptying the queue if you're doing it twice - I'm not sure that's the least unexpected behavior?

How about just doing `queue = []` in the `drainQueue` function - then there's a notion of one queue, just as if you had shifted them off.

Otherwise it's an overall elegant implementation.

Reply:
> This would mean emptying the queue if you're doing it twice - I'm not sure that's the least unexpected behavior?

Ah yes, unintended :)